sklearn/svm/tests/test_svm.py

```diff
@@ -560,6 +560,19 @@ def test_sparse_precomputed():
         assert "Sparse precomputed" in str(e)
 
 
+def test_sparse_fit_support_vectors_empty():
+    # Regression test for #14893
+    X_train = sparse.csr_matrix([[0, 1, 0, 0],
+                                 [0, 0, 0, 1],
+                                 [0, 0, 1, 0],
+                                 [0, 0, 0, 1]])
+    y_train = np.array([0.04, 0.04, 0.10, 0.16])
+    model = svm.SVR(kernel='linear')
+    model.fit(X_train, y_train)
+    assert model.support_vectors_.data.size == 0
```

Comment:
`model.support_vectors_.size` is enough (same below)

Reply:
I made the change. Please let me know if anything else can be improved!